api/src/opentrons/protocol_api/core/engine/instrument.py

```diff
@@ -1361,10 +1358,12 @@ def transfer_with_liquid_class(  # noqa: C901
             return_tip: If `True`, return tips to the tip rack location they were picked up from,
                         otherwise drop in `trash_location`
             keep_last_tip: When set to `True`, do not drop the final tip used in the transfer.
+            selected_tips: If provided, transfer will pick up the tips in the order given. If this
+                           is less than the amount of tips needed, an error will be raised.
         """
-        if not tip_racks:
+        if not tip_racks and not selected_tips:
             raise RuntimeError(
-                "No tipracks found for pipette in order to perform transfer"
+                "No tipracks or tips found for pipette in order to perform transfer"
```

Comment:
[🧐](https://opentrons.atlassian.net/wiki/spaces/RPDO/pages/980746400/Opentrons+UX+Style+Guide#Nouns)

```suggestion
                "No tip racks or tips found for pipette in order to perform transfer"
```